basisopt/basis/zetatools.py

```diff
@@ -77,6 +76,39 @@ def config_to_string(conf: Configuration) -> str:
         new_string = f"{v}{data.INV_AM_DICT[ix]}"
         value_string += new_string
     return value_string
+
+def string_to_config(string: str) -> Configuration:
+    """converts a configuration string, e.g. '4s3p2d1f',
+       to a Configuration dictionary 
+    """
+    config = {}
+    current_size = 0
+    for c in string:
+        try:
+            n = int(c)
+            current_size *= 10
```

Comment:
You are assuming that there are fewer than 10 functions per angular momentum here